homeassistant/components/netgear/__init__.py

```diff
@@ -52,14 +55,37 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         configuration_url=f"http://{entry.data[CONF_HOST]}/",
     )
 
+    async def async_update_data():
+        """Fetch data from the router."""
+        await router.async_update_device_trackers()
```

Comment:
The coordinator will only refresh automatically if there are registered listeners, ie entities. There may be 0 entities for the user when setting up the integration. How do we solve that so that the coordinator still scans for new devices?

Reply:
A first scan is done inside the init.py file in async_setup_entry await coordinator.async_config_entry_first_refresh()
That will always find some devices, at least the device running HomeAssistant schould be connected to the router so it schould find at least that device and create the entity for that.
Since there are then entities enabled, new devices can be discovered.

I can make a note in the documentation that at least one entity needs to be kept enabled in order for the integration to find new devices.

Reply:
Moreover if at some point the integration found a device, it will set it up again on a next restart using these lines:
https://github.com/home-assistant/core/blob/e147c05567b1675e1a4c030bf1a4dd004453a72f/homeassistant/components/netgear/router.py#L172-L193

Therefore if for instance a phone was tracked at some point and homeassistant restarts, the entity for the phone tracker will be setup again, even if the device is at that point not connected to the router and cannot be found by the router.

Reply:
I don't think we should rely on the user not disabling entities to still keep the integration working.

The `ScannerEntity` does no longer support creating a device registry entry with the release of tomorrow so we probably need to fix that restore code.

Reply:
We could register a listener before doing the first refresh to make sure that the coordinator keeps refreshing.

Reply:
@MartinHjelmare for each device there are also 3 SensorEntities registered (Link rate, Link Type, Signal Strength). Will those still ensure the device registry entry will be made or not?
Do you have any sugestion how I could fix the restore code?

Reply:
I just ensured that the coordinator keeps updating by registering a dummy listener.
@MartinHjelmare is this the propper implementation you ment?

Reply:
If there are entities from another domain than device_tracker attached to the device, the existing device will not be removed from the device registry when setting up the device tracker integration.

We should make sure that the netgear ScannerEntity doesn't implement the `device_info` property but instead the `mac_address` property. It also doesn't need to set a unique id as that will be set by the ScannerEntity class and will be the mac address. This will make sure the entity is linked up to the correct device via the mac address and still follow the new design rules.

Reply:
Done, I made sure the unique_id and device_info are no longer set in the ScannerEntity. The mac_address property was already implemented.